CollapseLauncher/Classes/CachesManagement/Honkai/Fetch.cs

```diff
@@ -34,57 +34,47 @@ private async Task<List<CacheAsset>> Fetch(CancellationToken token)
                 .SetAllowedDecompression(DecompressionMethods.None)
                 .Create();
 
-            // Use HttpClient instance on fetching
-            using Http httpClient = new Http(true, 5, 1000, _userAgent, httpClientNew);
-            try
-            {
-                // Subscribe the event listener
-                httpClient.DownloadProgress += _httpClient_FetchAssetProgress;
+            // Use a new DownloadClient for fetching
+            DownloadClient downloadClient = DownloadClient.CreateInstance(httpClientNew);
 
-                // Build _gameRepoURL from loading Dispatcher and Gateway
-                await BuildGameRepoURL(token);
+            // Build _gameRepoURL from loading Dispatcher and Gateway
+            await BuildGameRepoURL(downloadClient, token);
 
-                // Iterate type and do fetch
-                foreach (CacheAssetType type in Enum.GetValues<CacheAssetType>())
+            // Iterate type and do fetch
+            foreach (CacheAssetType type in Enum.GetValues<CacheAssetType>())
+            {
+                // Skip for unused type
+                switch (type)
                 {
-                    // Skip for unused type
-                    switch (type)
-                    {
-                        case CacheAssetType.Unused:
-                        case CacheAssetType.Dispatcher:
-                        case CacheAssetType.Gateway:
-                        case CacheAssetType.General:
-                        case CacheAssetType.IFix:
-                        case CacheAssetType.DesignData:
-                        case CacheAssetType.Lua:
-                            continue;
-                    }
+                    case CacheAssetType.Unused:
```

Comment:
Do we need to download the unused cache? If yes then remove the comment at L#46

Reply:
This is used to ignore the other types than Data, Event and AI (Asset Index).
Also, the reason why there are lots of Enum variant is that due to ``CacheAssetType`` being used for HSR and they have different types of data as well.